File: examples/e2e-travel-concierge-py/agent/tools/search_flights.py
# search_flights -- requires read:catalog. In the delegation demo, Research
# only gets this through the delegated child credential, not its base
# credential. Python port of ../../.../searchFlights.ts.

from __future__ import annotations

from typing import Optional

from config import TOOLS
from personas.types import Persona
from agent.tools.protected_call import ProtectedResult, call_protected_tool


def search_flights(persona: Persona, origin: str, destination: str, date: Optional[str] = None) -> ProtectedResult:
    return call_protected_tool(persona, TOOLS["SEARCH"], {"origin": origin, "destination": destination, "date": date or ""})

File: examples/e2e-travel-concierge-hosted/agent/tools/searchFlights.ts
// search_flights — requires read:catalog. In the delegation demo, Research only
// gets this through the delegated child credential, not its base credential.
import { TOOLS } from '../../config.js';
import type { Persona } from '../../personas/types.js';
import { callProtectedTool, type ProtectedResult } from './protectedCall.js';

export interface SearchFlightsArgs {
  origin: string;
  destination: string;
  date?: string;
}

export function searchFlights(persona: Persona, args: SearchFlightsArgs): Promise<ProtectedResult> {
  return callProtectedTool(persona, TOOLS.SEARCH, {
    origin: args.origin,
    destination: args.destination,
    date: args.date ?? '',
  });
}
